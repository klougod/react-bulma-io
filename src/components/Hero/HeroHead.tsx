import React, { ReactNode } from 'react'

interface HeroHeadProps {
  className?: string
  children?: ReactNode
}
export const HeroHead = ({ className, children }: HeroHeadProps) => (
  <div className={`hero-head ${className}`}>{children}</div>
)

HeroHead.defaultProps = {
  className: ''
}

export default HeroHead
