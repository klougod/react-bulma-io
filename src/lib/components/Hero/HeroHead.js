import React from 'react'
import PropTypes from 'prop-types'

export const HeroHead = ({ className, children }) => (
  <div className={`hero-head ${className}`}>
    { children }
  </div>
)

HeroHead.defaultProps = {
  className: ''
}

HeroHead.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default HeroHead