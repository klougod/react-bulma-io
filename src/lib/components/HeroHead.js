import React from 'react'
import PropTypes from 'prop-types'

export const HeroHead = props => {

  const {
    className,
    children
  } = props

  return (
    <div className={`hero-head ${className}`}>
      { children }
    </div>
  )
}

HeroHead.defaultProps = {
  className: ''
}

HeroHead.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default HeroHead