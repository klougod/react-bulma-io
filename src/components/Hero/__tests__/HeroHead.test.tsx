import React from 'react'
import Enzyme from 'enzyme'

import HeroHead from '../HeroHead'

describe('HeroHead component test', function () {
  it('should render without crash without props', function () {
    const heroHeadMock = Enzyme.shallow(<HeroHead />)
    expect(heroHeadMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const heroHeadMockProps = {
      className: 'heroHead CLass',
    }
    const heroHeadMock = Enzyme.shallow(<HeroHead {...heroHeadMockProps} />)
    expect(heroHeadMock).toMatchSnapshot()
  })
})
