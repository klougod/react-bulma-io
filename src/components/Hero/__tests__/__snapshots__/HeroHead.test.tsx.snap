// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroHead component test should render without crash with props 1`] = `
<div
  className="hero-head heroHead CLass"
/>
`;

exports[`HeroHead component test should render without crash without props 1`] = `
<div
  className="hero-head "
/>
`;
